Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Order.lean

```diff
@@ -6,28 +6,94 @@ Authors: Frédéric Dupuis
   statements for `a * star a` and `star a * a`.
+* `CstarRing.norm_le_norm_of_nonneg_of_le`: in a non-unital C⋆-algebra, if `0 ≤ a ≤ b`, then
+  `‖a‖ ≤ ‖b‖`.
+* `CstarRing.conjugate_le_norm_smul`: in a non-unital C⋆-algebra, we have that
+  `star a * b * a ≤ ‖b‖ • (star a * a)` (and a primed version for the `a * b * star a` case).
 
 ## Tags
 
 continuous functional calculus, normal, selfadjoint
 -/
 
+open scoped NNReal
+
+namespace Unitization
+
+variable {A : Type*} [NonUnitalNormedRing A] [CompleteSpace A] [Nontrivial A]
+  [PartialOrder A] [StarRing A] [StarOrderedRing A] [CstarRing A] [NormedSpace ℂ A] [StarModule ℂ A]
+  [SMulCommClass ℂ A A] [IsScalarTower ℂ A A]
+
+instance instPartialOrder : PartialOrder (Unitization ℂ A) := CstarRing.spectralOrder _
+
+instance instStarOrderedRing : StarOrderedRing (Unitization ℂ A) := CstarRing.spectralOrderedRing _
+
+lemma inr_le_iff (a b : A) (ha : IsSelfAdjoint a := by cfc_tac)
+    (hb : IsSelfAdjoint b := by cfc_tac) :
+    (a : Unitization ℂ A) ≤ (b : Unitization ℂ A) ↔ a ≤ b := by
+  -- TODO: prove the more general result for star monomorphisms and use it here.
+  rw [← sub_nonneg, ← sub_nonneg (a := b), StarOrderedRing.nonneg_iff_spectrum_nonneg (R := ℝ) _,
+    ← inr_sub ℂ b a, ← Unitization.quasispectrum_eq_spectrum_inr' ℝ ℂ]
+  exact StarOrderedRing.nonneg_iff_quasispectrum_nonneg _ |>.symm
+
+@[simp, norm_cast]
+lemma inr_nonneg_iff {a : A} : 0 ≤ (a : Unitization ℂ A) ↔ 0 ≤ a := by
+  by_cases ha : IsSelfAdjoint a
+  · exact inr_zero ℂ (A := A) ▸ inr_le_iff 0 a
+  · refine ⟨?_, ?_⟩
+    all_goals refine fun h ↦ (ha ?_).elim
+    · exact isSelfAdjoint_inr (R := ℂ) |>.mp <| .of_nonneg h
+    · exact .of_nonneg h
+
+end Unitization
+
+section generic
+
+variable {A : Type*} [TopologicalSpace A] [Ring A] [StarRing A] [PartialOrder A]
+  [Algebra ℂ A] [ContinuousFunctionalCalculus ℂ (IsStarNormal : A → Prop)]
+  [NonnegSpectrumClass ℝ A]
+
+lemma IsSelfAdjoint.coe_mem_spectrum_complex {a : A} {x : ℝ} (ha : IsSelfAdjoint a := by cfc_tac) :
+    (x : ℂ) ∈ spectrum ℂ a ↔ x ∈ spectrum ℝ a := by
+  simp [← ha.spectrumRestricts.algebraMap_image]
+
+lemma coe_mem_spectrum_real_of_nonneg {a : A} {x : ℝ≥0} (ha : 0 ≤ a := by cfc_tac) :
+    (x : ℝ) ∈ spectrum ℝ a ↔ x ∈ spectrum ℝ≥0 a := by
+  simp [← (SpectrumRestricts.nnreal_of_nonneg ha).algebraMap_image, Set.mem_image,
+    NNReal.algebraMap_eq_coe]
+
+end generic
+
+section Cstar_unital
+
 variable {A : Type*} [NormedRing A] [StarRing A] [CstarRing A] [CompleteSpace A]
 variable [NormedAlgebra ℂ A] [StarModule ℂ A] [PartialOrder A] [StarOrderedRing A]
 
+open ComplexOrder in
+instance CstarRing.instNonnegSpectrumClassComplexUnital : NonnegSpectrumClass ℂ A where
+  quasispectrum_nonneg_of_nonneg a ha x := by
+    rw [mem_quasispectrum_iff]
+    refine (Or.elim · ge_of_eq fun hx ↦ ?_)
+    obtain ⟨y, hy, rfl⟩ := (IsSelfAdjoint.of_nonneg ha).spectrumRestricts.algebraMap_image ▸ hx
+    simpa using spectrum_nonneg_of_nonneg ha hy
```

Comment:
Can you please move this to the `Instances` file, next to the corresponding `NonnegSpectrumClass ℝ A` instance?